### PR TITLE
Feature/template builder collaborators page

### PR DESCRIPTION
**Pull request by fraserclark**
## Description
Issue
https://github.com/CDLUC3/dmsp_frontend_prototype/issues/166

Created a non-functional "Manage Access/Collaborators" page for template builder. 

This will definitely require a second pass on componentising the "People" list but we are using this style in other places, so I will make it a component when I have a few of them built so it can be flexibly used 

Fixes # (issue)

## Type of change
Please delete options that are not relevant

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
I have used an AXE accessibility test plugin, cross browser and device testing manually.

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md and added documentation if necessary
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have completed manual or automated accessibility testing for my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: styles/_elements.scss [hunk 1]
```diff
@@ -76,13 +76,15 @@
 }
 
 // Forms
-
+/*
+// REMOVED THIS AS WAS CONFLICTING - Unclear where it is used
```
Comment (fraserclark) [inline styles/_elements.scss:80]: I have a conflicting style with a large margin top removed for now

File: app/template/[templateId]/access/page.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            <div className={styles.AccessSectionHeader}>
+              <h3 id="external-access-heading">External People</h3>
+            </div>
+            <div>
+              <p>Share this template with people outside NSF</p>
+              <div className={styles.externalPeopleList}>
+                {externalPeople.length > 0 ? (
+                  <ul className={styles.peopleList} role="list">
+                    {externalPeople.map((person) => (
+                      <li key={person.id} className={styles.personItem}>
```
Comment (fraserclark) [inline app/template/[templateId]/access/page.tsx:110]: This <li> will likely become a component later in life - but for now kept it simple

File: app/template/[templateId]/access/__tests__/page.spec.tsx
Comment (fraserclark) [inline app/template/[templateId]/access/__tests__/page.spec.tsx:1]: Setup bare minimum boilerplate test - once we start adding real functionality we can improve this

File: app/template/[templateId]/access/page.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+
+  return (
+    <div>
+      <PageHeader
+        title={"Manage Access"}
+        description=""
+        showBackButton={true}
+        breadcrumbs={
+          <Breadcrumbs>
```
Comment (jupiter007) [inline app/template/[templateId]/access/page.tsx:73]: @fraserclark , what do you think about the way the breadcrumb looks on a mobile device?

Just for fun, I added some more items to the breadcrumb, and noticed it starts to get really squished, with the arrow not aligning vertically, but being located at the bottom of the wrapping text.

image

File: app/template/[templateId]/access/page.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+              <TextField>
+                <Label>Email</Label>
+                <Input
+                  type="email"
+                  value={""}
+
+                  required
+                  aria-required="true"
+                  aria-describedby="email-hint"
```
Comment (jupiter007) [inline app/template/[templateId]/access/page.tsx:155]: L155: @fraserclark I don't see the corresponding element with id="email-hint"

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,6 @@
+### Added
+- Template Builder: Access Page
```
Comment (jupiter007) [inline CHANGELOG.md:2]: @fraserclark could you add the ticket number alongside your entry, as well as linking to it from your PR's description section?

File: app/template/[templateId]/access/TemplateAccessPage.module.scss [hunk 6]
```diff
@@ -0,0 +1,65 @@
+
+.AccessSectionHeader {
```
[the remark this answers is not in the document]
Comment (fraserclark) [reply app/template/[templateId]/access/TemplateAccessPage.module.scss:2]: Yeah probably right, I will make more generic and fix the mistaken case

**Comment by fraserclark** [comment]
I have updated these.

Section is now in element.scss - has somewhat standardised spacing we can re-use 
Breadcrumbs - its always an issue on mobile, I want to avoid making it look like "navigation", so on smaller devices wrote some code to hide the "current page" and show the last 2 "parent pages". Additionally added some CSS truncation to fit the space, will  be fully read by screen readers

**Comment by jupiter007** [comment]
> I have updated these.
> 
> Section is now in element.scss - has somewhat standardised spacing we can re-use Breadcrumbs - its always an issue on mobile, I want to avoid making it look like "navigation", so on smaller devices wrote some code to hide the "current page" and show the last 2 "parent pages". Additionally added some CSS truncation to fit the space, will be fully read by screen readers

Awesome @fraserclark. Thanks so much. It looks good.

**Review by jupiter007 — APPROVED** [review]

**Review by briri — APPROVED** [review]
LGTM

**Pull request merged**